File: web/types/movie.ts
export interface Movie {
  id: number;
  title: string;
  overview: string | null;
  posterPath: string | null;
  backdropPath: string | null;
  voteAverage: number;
  releaseDate: string | null;
}

export interface MovieDetails {
  movie: Movie;
  trailerKey: string | null;
}

File: api/src/MovieSearchCase.Domain/Entities/Movie.cs
namespace MovieSearchCase.Domain.Entities;

public class Movie
{
    public required int Id { get; init; }

    public required string Title { get; init; }

    public string? Overview { get; init; }

    public string? PosterPath { get; init; }

    public string? BackdropPath { get; init; }

    public double VoteAverage { get; init; }

    public DateOnly? ReleaseDate { get; init; }
}
